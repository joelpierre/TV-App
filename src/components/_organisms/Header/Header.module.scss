.Header {
  width: 100%;
  position: relative;
  background-color: $color-tint-psi;
  padding-top: spacing(2);
  padding-bottom: spacing(2);
  z-index: 100;

  &__container {
    @include max-width-container;
  }

  &__logo {
    max-width: 150px;
  }
}
