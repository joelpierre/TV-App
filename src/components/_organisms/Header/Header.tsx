import classNames from 'classnames';
import React from 'react';

import styles from './Header.module.scss';
import Anchor from '@atoms/Anchor';
import { EPageHref } from 'common/types/enums';

interface IFooterProps {
  className?: string;
}

const Header: React.FunctionComponent<IFooterProps> = ({
  className
}) => {
  return (
    <header className={classNames(styles.Header, className)}>
      <div className={styles.Header__container}>
        <Anchor href={EPageHref.Home} as="/">
          <img
            data-testid="logo"
            className={styles.Header__logo}
            src="/images/tvm-header-logo.png"
            alt="Logo for TV Maze"
            width="200"
            loading="lazy"
          />
        </Anchor>
      </div>
    </header>
  );
};

export default Header;
