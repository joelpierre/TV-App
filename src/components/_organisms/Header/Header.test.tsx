import React from 'react';
import { render, RenderResult } from 'test-utils';

import Header from './Header';

const setup = (props = {}): RenderResult => {
  const setupProps = { ...props };
  return render(<Header {...setupProps} />);
};

describe('<Header/>', () => {
  let wrapper: RenderResult;

  beforeEach(() => {
    wrapper = setup();
  });

  it('Should render without crashing', () => {
    expect(wrapper.container).not.toBeEmptyDOMElement();
  });

  it('should render logo', () => {
    expect(wrapper.container.getElementsByTagName('img')).toHaveLength(1);
    expect(wrapper.queryByTestId('logo')).toBeTruthy();
  });

  it('should match snapshot', () => {
    expect(wrapper.asFragment()).toMatchSnapshot();
  });
});
