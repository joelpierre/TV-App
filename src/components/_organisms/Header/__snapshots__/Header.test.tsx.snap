// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header/> should match snapshot 1`] = `
<DocumentFragment>
  <header
    class="Header"
  >
    <div
      class="Header__container"
    >
      <a
        class="Anchor"
        href="/"
      >
        <img
          alt="Logo for TV Maze"
          class="Header__logo"
          data-testid="logo"
          loading="lazy"
          src="/images/tvm-header-logo.png"
          width="200"
        />
      </a>
    </div>
  </header>
</DocumentFragment>
`;
